### DISCUSSION TITLE
FeatureCollisionAttack doesn't support pytorch

### DISCUSSION THREAD

**Discussion by cnut1648** [Trusted-AI/adversarial-robustness-toolbox#1555] [Q&A]
**Describe the bug**
Hi, I think currently `FeatureCollisionAttack` doesn't support pytorch. 

**To Reproduce**
I have the following script
```python
import torch
from art.attacks.poisoning.feature_collision_attack import FeatureCollisionAttack
from art.estimators.classification import PyTorchClassifier
from torch import nn
from torch import optim
from torchvision.datasets import MNIST
from torchvision.transforms import transforms

transform = transforms.Compose([
    transforms.ToTensor(),
    ])

mnist = MNIST(root=".", download=True, transform=transform, train=True)

mnist_loader = torch.utils.data.DataLoader(dataset=mnist,
                                           batch_size=32,
                                           shuffle=True,
                                           num_workers=1)

model = nn.Sequential(
    nn.Flatten(),
    nn.Linear(28 * 28, 512),
    nn.Linear(512, 10)
)
optim = optim.Adam(model.parameters(), lr=1e-5)
artmodel = PyTorchClassifier(
    model=model, loss=nn.CrossEntropyLoss(),
    input_shape=(32, 1, 28, 28), nb_classes=10,
    optimizer=optim, device_type="cpu"
)
for img, label in mnist_loader:
    logits = model(img)
    poison = FeatureCollisionAttack(artmodel, img[:5], decay_coeff=0.1,
                 feature_layer='1_Linear(in_features=784, out_features=512, bias=True)',
                 max_iter=200, verbose=False)
    poison.poison(img)
```
First of all, `FeatureCollisionAttack` can't be init as `PyTorchClassifier` is not `KerasClassifier` so it won't pass the assert in `feature_collision_attack.py` line 100. Even if I comment it out, in `feature_collision_attack.py` line 114 it would return only one tensor but it seems that it will return two.

**System information (please complete the following information):**
 - OS: Manjaro
 - Python version: Python 3.7
 - ART version or commit number: 1.9.1
 - TensorFlow / Keras / PyTorch / MXNet version: Pytorch 1.8.1

**Comment by beat-buesser** [comment]
Hi @cnut1648 Thank you very much for using ART! Yes, that correct, so far `FeatureCollisionAttack` only supports Keras models with `KerasClassifier`. We are working on adding support for PyTorch in ART 1.10 with PR #1435. It should be ready to merge, if you have time please pull that feature branch to test the new PyTorch support and let me know what you find.

We agree, the pattern in line `114` is not good, we are looking to replace the initialisation of the placeholders with a better pattern soon.
